### DISCUSSION TITLE
Starting map

### DISCUSSION THREAD

**Discussion by rockettpc** [itzg/docker-minecraft-bedrock-server#375] [Q&A]
Is there an environment variable or command we can run before starting the server so that players have a starting map?

**Comment by itzg** [comment]
> Is there an environment variable or command we can run before starting the server so that players have a starting map?

Like a pre generated or downloaded world?

**Reply by rockettpc** [reply]
For example, when spinning up a new Minecraft bedrock server in docker, and letting the default install generate a new world.

**Reply by itzg** [reply]
> letting the default install generate a new world

That's what always happens. Please clarify what behavior is not working as expected.

**Reply by rockettpc** [reply]
When I spin up a new Minecraft server the players join the world completely empty handed, no starting map.

**Reply by itzg** [reply]
Sorry, now I understand what you mean. Let's see if anyone else has an answer since that's not specific to the docker image.

**Reply by rockettpc** [reply]
Hey no problem, thanks for all the info

**Reply by BanditTech** [reply]
I am also trying to figure out how to set this in the server.properties
Been googling for a solution for a while now :P
Just realized this is on the github for java edition and not bedrock, do not think this option even exists on java version.

**Reply by itzg** [reply]
Correct, I'm 99% certain that feature doesn't exist for Java edition. Maybe should move this discussion to the docker bedrock repository.

**Reply by BanditTech** [reply]
:D Thank you for moving the conversation, I went to the proper repo and already found this same thread here. 

After doing lots and lots of searching, and still unable to find any information about how to do this as a server setting.
From what I have gathered, its an option given to you when creating a server from the client UI.
I assume this would mean there is some parity to the server.properties file.
Based on the formatting of the other options, I assume it would be `starting-map=true`

There are other missing options such as `bonus chest`, `show coordinates`, `motd`

I am trying to make a simple world for my son that starts him out in a more forgiving play through.
He lost his home on one occasion and it was a heartbreaking lesson to have him learn!

I will be trying to figure out these options and updating you if I can trial and error them to work.

**Reply by BanditTech** [reply]
unfortunately none of these options have any effect in the `server.properties` file

**Reply by BanditTech** [reply]
as an operator, I can use `/gamerule showcoordinates true`
but that still leaves the missing bonus chest and starting map.

**Reply by itzg** [reply]
Hmm, no, I'm not seeing a server property for starting-map

https://minecraft.wiki/w/Server.properties

Since [it seems to be a UI only option](https://minecraft.wiki/w/Map#Starting_map), perhaps a world could be generated with the client and then transferred to the server data directory.

**Reply by BanditTech** [reply]
You gave me the idea to instead edit the level.dat file. I was able to do it with Universal Minecraft Editor, its part of its free base functionality.

to re-enable achievements:
`hasBeenLoadedInCreative`

and these are for the relevant options discussed above:
`startWithMapEnabled`
`bonusChestEnabled`
`showcoordinates`

another option that might be useful for making kid friendly server:
`keepinventory`
`respawnblocksexplode`

I used this information to produce a new world, with the options enabled at start:
* Produce a new level by stopping the container, and edit server.properties field `level-name`
* Start the container, this will produce the newly named level.
* Stop the container, and copy the newly created level from your worlds folder.
* Open the local copy of the level, and edit the relevant fields.
* Replace the level on the server with your local copy.
* Start the container.

**Reply by rockettpc** [reply]
Sweet I will have to try this! Thanks!

**Reply by rockettpc** [reply]
I copied a map from my wifes nintendo switch, heres how.

Subscribed to Realms and uploaded the map. This lets you access your Minecraft worlds from any device. You can even play together with friends cross-platform, but its a monthly sub, once its copied though you can un-sub. 

Downloaded the map on my PC. Once it was uploaded to Realms, I could easily download it to my windows VM.
Exported the map from the Minecraft UI. This creates a file that I can import onto other servers.

That's as far as I've gotten for now, but I'm planning to tackle importing it as the main map on our Docker server next. It'll likely be a Bedrock server only, just because that's how the Switch version works. But, the ultimate goal is to get it working with our Geyser Paper MC server so friends and family, regardless of platform, can join in the fun on my wife's map.
